.github/workflows/release.yaml

```diff
@@ -0,0 +1,67 @@
+name: release
+
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - 'main'
+    tags:
+      - 'v*'
+  pull_request:
+    branches:
+    - main
+
+jobs:
+  goreleaser:
+    name: goreleaser
+    runs-on: ubuntu-latest
+    steps:
+    - name: Check out code into the Go module directory
+      uses: actions/checkout@v3
+      with:
+        fetch-depth: 0
+
+    - name: Install Go
+      uses: actions/setup-go@v3
+      with:
+        go-version: ~1.18
```

Comment:
Since you're already using actions/checkout@v3 you could also use the `go-version-file`, like this: https://github.com/operator-framework/operator-registry/blob/614d6a955dc0d7d341dfa71215b373bf401864c3/.github/workflows/build.yaml#L18

Reply:
It would probably be better if we do all the files at the same time.  It does make it theoretically easier to pivot go versions, but at the cost of always needing to do a checkout of the code first.

Reply:
HAHA.  Cached pages are awful.  Thought I'd lost my comment, but then I went back against the idea myself.  Seems like -- by accident of horrible internet -- we arrived at consensus. :)